### ISSUE TITLE
Typing 'b' or 'h' into the Diff Filter Files text box doesn't insert the character

### ISSUE THREAD

**Issue by wplpgr** [gitextensions/gitextensions#5065]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
Typing either "b" or "h" as a character in the "Diff" "Filter files..." text box causes the "Blame" or "History" windows to appear. 

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

1. Open a repo
1. Select a commit with a parent
1. Go to the "Diff" tab in the bottom panel
1. Place the text entry cursor into the "Filter files..." text box
1. Type "b" or "h"

**What is the expected behavior?**
It should let me type the "b" and "h" as characters in the filter string.

**Environment you encounter the issue:**

- GitExtensions version: master fa860cdfda35d663a90d0a9955a6f95854191ba6
- GIT version: git-for-windows 2.17.1.windows.2
- OS version: Windows 10 (1703)

**Did this work in previous version of GitExtensions (which)?**
No idea, sorry.

**Comment by gerhardol** [comment]
Not in 2.51.02

**Comment by drewnoakes** [comment]
Bisect shows this was introduced in 03e16f5aa64a4a8c3798221637e6b46eefc8b2c2.

```
03e16f5aa64a4a8c3798221637e6b46eefc8b2c2 is the first bad commit
commit 03e16f5aa64a4a8c3798221637e6b46eefc8b2c2
Author: Michael Seibt <mstv@gmx.net>
Date:   Thu Feb 22 23:47:24 2018 +0100

    F3 opens the external diff viewer unless searching text.
```

@mstv is this something you could take a look at?

**Comment by wplpgr** [comment]
@mstv to be fair, I think this was the first time I'd noticed it - it was just (un)lucky that I decided to use it to find all the C++ header files that had changed in a commit.

Oh, one other thing I've managed to reproduce (but may be directly related to this): if there was no file selected in the diff, typing 'b' in the filter field causes gitextensions to crash with a NullReferenceException. If it's a different bug, I can raise it separately.

Void blameToolStripMenuItem_Click(System.Object, System.EventArgs) @ GitUI.CommandsDialogs.RevisionDiff

   at GitUI.CommandsDialogs.RevisionDiff.blameToolStripMenuItem_Click(Object sender, EventArgs e) in C:\Users\pr\src\local\gitextensions\GitUI\CommandsDialogs\RevisionDiff.cs:line 457
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.PerformClick()
   at GitUI.CommandsDialogs.RevisionDiff.ExecuteCommand(Int32 cmd) in C:\Users\pr\src\local\gitextensions\GitUI\CommandsDialogs\RevisionDiff.cs:line 106
   at ResourceManager.GitExtensionsControl.ProcessCmdKey(Message& msg, Keys keyData) in C:\Users\pr\src\local\gitextensions\ResourceManager\GitExtensionsUserControl.cs:line 101
   at System.Windows.Forms.Control.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.PreProcessMessage(Message& msg)
   at System.Windows.Forms.Control.PreProcessControlMessageInternal(Control target, Message& msg)
   at System.Windows.Forms.Application.ThreadContext.PreTranslateMessage(MSG& msg)
